### PR TITLE
improve facet_grid and facet_wrap documentation

### LINKED ISSUE #3290: Improve facet_*() documentation
There are several opportunities to improve the [facet_wrap()](https://ggplot2.tidyverse.org/reference/facet_wrap.html) and [facet_grid()](https://ggplot2.tidyverse.org/reference/facet_grid.html) documentation. This is a documentation issue intended for the tidyverse developer day (#3267).

- In `facet_grid()` and `facet_wrap()`, I'm not sure it's the best idea to illustrate the formula interface in the examples (is this encouraged syntax?) (#3181). I think these examples should possibly be removed.

- There is a confusing warning in the last example of `facet_wrap()` ("Suppressing axis rendering when strip.position = 'bottom' and strip.placement == 'outside'"). Perhaps a different example should be chosen or the warning explained?

- The documentation for the `labeller` argument in `facet_grid()` and `facet_wrap()` contains the old formula notation rather than the new `vars()` notation. I'm also not sure that the function that is passed has to inherit from "labeller" (this should be checked, I seem to remember ignoring this advice and having no problems). It also may be useful to mention `label_parsed()` as a way to get formatting into the facet labels, and that the default is `label_value()`.

- The [`labeller()` documentation](https://ggplot2.tidyverse.org/reference/labeller.html) should mention that it works in `facet_wrap()` in addition to `facet_grid()` (there are examples for both but only a link to `facet_grid()`).

- `facet_grid()` should link to `facet_wrap()` in the first paragraph (the reverse is true).

These are only suggestions, feel free to add to/debate!

### PR DESCRIPTION
**Pull request by andrasberczi**
Closes #3290

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -6,6 +6,8 @@ ggplot2 1.0.1
 * Improvements to order of colours and legend display for continuous colour
   scales extracted from colorbrewer palettes by `scale_*_distiller()` (@jiho, 1076)
 
+* Improve facet documentation (@paleolimbot, #3290)
```
Comment (hadley) [inline NEWS]: This should go in `NEWS.md`

File: R/facet-grid-.r [hunk 2]
```diff
@@ -6,6 +6,7 @@ NULL
 #' `facet_grid()` forms a matrix of panels defined by row and column
 #' faceting variables. It is most useful when you have two discrete
 #' variables, and all combinations of the variables exist in the data.
+#' When having only one variable to facet use [facet_wrap()].
```
Comment (hadley) [inline R/facet-grid-.r]:
```suggestion
#' If you have only one variable with many levels, try [facet_wrap()].
```

File: R/facet-grid-.r [hunk 3]
```diff
@@ -28,12 +29,13 @@ NULL
 #' @param labeller A function that takes one data frame of labels and
 #'   returns a list or data frame of character vectors. Each input
 #'   column corresponds to one factor. Thus there will be more than
-#'   one with formulae of the type `~cyl + am`. Each output
+#'   one with formulae of the type `vars(cyl, am)`. Each output
```
Comment (hadley) [inline R/facet-grid-.r]: Also need to update "formulae"

File: R/facet-grid-.r [hunk 4]
```diff
@@ -28,12 +29,13 @@ NULL
...
 #'   column gets displayed as one separate line in the strip
 #'   label. This function should inherit from the "labeller" S3 class
-#'   for compatibility with [labeller()]. See
-#'   [label_value()] for more details and pointers to other
-#'   options.
+#'   for compatibility with [labeller()]. You can use different labeling 
+#'   functions to have different kind of labels, for example use [label_parsed()] for
```
Comment (hadley) [inline R/facet-grid-.r]:
```suggestion
#'   functions for different kind of labels, for example use [label_parsed()] for 
```

**Comment by andrasberczi** [comment]
I just pushed the changes, but forgot to do `document()`, one moment and I am going to fix it :)

**Comment by andrasberczi** [comment]
Done

**Comment by hadley** [comment]
Thanks @andrasberczi !

**Pull request merged**